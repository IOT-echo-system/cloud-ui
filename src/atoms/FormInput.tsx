import type {TextFieldProps} from '@mui/material'
import {TextField} from '@mui/material'

export type FormInputType = TextFieldProps

export const FormInput: React.FC<FormInputType> = props => {
  return <TextField variant={props.variant ?? 'outlined'} {...props} />
}
