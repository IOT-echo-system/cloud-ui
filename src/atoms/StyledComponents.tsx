import {styled, Button as MUIButton, Stack} from '@mui/material'

export const Button = styled(MUIButton)({textTransform: 'none'})

export const FormContainer = styled('form')(({theme}) => ({
  background: theme.palette.background.paper,
  padding: theme.spacing(4),
  boxShadow: theme.shadows[1],
  borderRadius: theme.spacing(1),
  width: theme.spacing(60),
  '&>*': {
    margin: theme.spacing(1.5)
  },
  '&.light': {
    border: `1px solid ${theme.palette.grey[300]}`
  },
  '&.dark': {
    color: theme.palette.common.white,
    border: `1px solid ${theme.palette.grey[700]}`
  }
}))

export const CenteredContainer = styled(Stack)(() => ({
  margin: 'auto'
}))
