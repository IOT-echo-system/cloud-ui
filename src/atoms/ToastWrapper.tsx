import type {PropsWithChildren} from 'react'
import React, {createRef} from 'react'
import type {SnackbarKey} from 'notistack'
import {SnackbarProvider} from 'notistack'
import {IconButton} from '@mui/material'
import {Close} from '@mui/icons-material'
import {styled} from '@mui/material/styles'

const Toast = styled(SnackbarProvider)(({theme}) => ({
  width: theme.spacing(48)
}))

export const ToastWrapper: React.FC<PropsWithChildren> = ({children}) => {
  const toastRef = createRef<SnackbarProvider>()
  const onClickDismiss = (key: SnackbarKey) => () => {
    toastRef.current?.closeSnackbar(key)
  }

  return (
    <Toast
      maxSnack={5}
      anchorOrigin={{horizontal: 'right', vertical: 'top'}}
      autoHideDuration={5000}
      ref={toastRef}
      aria-multiline={true}
      action={key => (
        <IconButton onClick={onClickDismiss(key)}>
          <Close />
        </IconButton>
      )}
    >
      {children}
    </Toast>
  )
}
