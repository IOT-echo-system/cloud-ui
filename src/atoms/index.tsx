export * from './StyledComponents'
export * from './FormInput'
