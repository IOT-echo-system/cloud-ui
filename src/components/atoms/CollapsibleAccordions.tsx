import React from 'react'
import {
  Accordion as MuiAccordion,
  AccordionDetails as MuiAccordionDetails,
  AccordionSummary,
  Typography
} from '@mui/material'
import {ExpandMore} from '@mui/icons-material'
import {styled} from '@mui/material/styles'

const Accordion = styled(MuiAccordion)(({theme}) => ({
  border: `1px solid ${theme.palette.divider}`,
  borderLeft: 0,
  borderRight: 0,
  '&:not(:last-child)': {
    borderBottom: 0
  },
  '&::before': {
    display: 'none'
  },
  '&.Mui-expanded': {
    background: theme.palette.grey[50]
  }
}))

const AccordionDetails = styled(MuiAccordionDetails)(({theme}) => ({
  padding: theme.spacing(2),
  borderTop: `1px solid ${theme.palette.grey[200]}`,
  background: theme.palette.grey[50]
}))

export type AccordionType = {
  details: React.JSX.Element
  summary: string
}
type CollapsibleAccordionPropsType = {accordions: AccordionType[]; header?: ReturnType<React.FC>}

export const CollapsibleAccordion: React.FC<CollapsibleAccordionPropsType> = ({accordions, header}) => {
  return (
    <div>
      {header}
      {accordions.map((accordion, index) => {
        return (
          <Accordion
            disableGutters
            key={`accordion-${index}`}
            square={index === 0 && !!header}
            defaultExpanded={index === 0}
          >
            <AccordionSummary expandIcon={<ExpandMore />}>
              <Typography>{accordion.summary}</Typography>
            </AccordionSummary>
            <AccordionDetails>{accordion.details}</AccordionDetails>
          </Accordion>
        )
      })}
    </div>
  )
}
