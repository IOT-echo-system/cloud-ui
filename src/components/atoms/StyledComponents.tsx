import type {BoxProps} from '@mui/material'
import {PropTypes} from '@mui/material'
import {Box, Stack, styled} from '@mui/material'
import {LoadingButton} from '@mui/lab'
import NextLink from 'next/link'
import Color = PropTypes.Color

export const Button = styled(LoadingButton)({textTransform: 'none'})

export const FormContainer = styled('form')(({theme}) => ({
  background: theme.palette.background.paper,
  padding: theme.spacing(1),
  boxShadow: theme.shadows[1],
  borderRadius: theme.spacing(1),
  margin: theme.spacing(1, 'auto'),
  justifyContent: 'center',
  width: `calc(100vw - ${theme.spacing(4)})`,
  '&>*': {
    margin: theme.spacing(1.5)
  },
  [theme.breakpoints.up('sm')]: {
    width: theme.spacing(60),
    padding: theme.spacing(4)
  }
}))

export const BoxContainer = styled(Box)<BoxProps>(({theme}) => ({
  background: theme.palette.background.paper,
  boxShadow: theme.shadows[1],
  borderRadius: theme.spacing(1),
  margin: theme.spacing(1),
  justifyContent: 'center',
  width: `calc(100vw - ${theme.spacing(4)})`,
  '&>*': {
    margin: theme.spacing(2)
  },
  [theme.breakpoints.up('sm')]: {
    width: theme.spacing(72),
    '&>*': {
      margin: theme.spacing(2.5)
    }
  },
  '&>hr': {
    margin: 0
  }
}))

export const CenteredContainer = styled(Stack)(({theme}) => ({
  margin: 'auto',
  minHeight: '100vh',
  justifyContent: 'center',
  width: '100vw',
  [theme.breakpoints.up('sm')]: {
    width: theme.spacing(72)
  }
}))

export const TopCenteredContainer = styled(Stack)(({theme}) => ({
  margin: theme.spacing(4, 'auto')
}))

type LinkPropsType = {disableUnderline?: boolean; color?: Color}
export const Link = styled(NextLink)<LinkPropsType>(({disableUnderline, color}) => ({
  textDecoration: 'none',
  color: color ?? 'black',
  '&:hover': {
    textDecoration: disableUnderline ? 'none' : 'underline'
  }
}))

export const PageContainer = styled(Stack)(({theme}) => ({
  margin: '0 auto',
  width: '95%',
  [theme.breakpoints.up('md')]: {
    width: '90%'
  },
  [theme.breakpoints.up('lg')]: {
    width: '80%'
  }
}))
