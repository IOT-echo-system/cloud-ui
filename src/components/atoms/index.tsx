export * from './StyledComponents'
export * from './FormInput'
export * from './ToastWrapper'
export * from './Loader'
export * from './CollapsibleAccordions'
export * from './Modal'
export * from './CopyRights'
export * from './Breadcrumbs'
export * from './ModalForm'
export * from './ModalFormSelect'
export * from './PolicyAllowed'
export * from './ConfirmationModal'
