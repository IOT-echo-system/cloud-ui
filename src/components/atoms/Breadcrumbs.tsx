import React from 'react'
import {Breadcrumbs as MuiBreadcrumbs, Typography} from '@mui/material'
import {Link} from './StyledComponents'
import {NavigateNext} from '@mui/icons-material'

type BreadcrumbsPropsType = {links: Array<{link: string; name: string}>; text: string}

export const Breadcrumbs: React.FC<BreadcrumbsPropsType> = ({links, text}) => {
  return (
    <MuiBreadcrumbs separator={<NavigateNext />}>
      {links.map(({link, name}) => (
        <Link href={link} key={link}>
          {name}
        </Link>
      ))}
      <Typography>{text}</Typography>
    </MuiBreadcrumbs>
  )
}
