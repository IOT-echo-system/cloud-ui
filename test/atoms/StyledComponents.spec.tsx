import {render} from '@testing-library/react'
import {Button, CenteredContainer, FormContainer} from '@/atoms/StyledComponents'

describe('Styled component test', () => {
  it('should match with the snapshot for Button', () => {
    const {container} = render(<Button />)

    expect(container).toMatchSnapshot()
  })

  it('should match with the snapshot for Centered Container', () => {
    const {container} = render(<CenteredContainer />)

    expect(container).toMatchSnapshot()
  })

  it('should match with the snapshot for FormContainer', () => {
    const {container} = render(<FormContainer />)

    expect(container).toMatchSnapshot()
  })
})
