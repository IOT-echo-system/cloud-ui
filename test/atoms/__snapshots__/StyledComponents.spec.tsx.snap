// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled component test should match with the snapshot for Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zxi1gz-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Styled component test should match with the snapshot for Centered Container 1`] = `
<div>
  <div
    class="MuiStack-root css-1edlbwa-MuiStack-root"
  />
</div>
`;

exports[`Styled component test should match with the snapshot for FormContainer 1`] = `
<div>
  <form
    class="css-yg4hvw"
  />
</div>
`;
